int64 marknumber
bool permanent
---

bool result
